internal/provider/resource_integration.go

```diff
@@ -405,33 +405,19 @@ func (r *integrationResource) Delete(ctx context.Context, req resource.DeleteReq
 
 	var apps []nestedApplication
 	state.Application.ElementsAs(ctx, &apps, false)
-	endpoints, offer, _, err := parseEndpoints(apps)
+	endpoints, _, _, err := parseEndpoints(apps)
 	if err != nil {
 		resp.Diagnostics.AddError("Provider Error", err.Error())
 		return
 	}
 
-	//If one of the endpoints is an offer then we need to remove the remote offer rather than destroying the integration
-	if offer != nil {
-		errs := r.client.Offers.RemoveRemoteOffer(&juju.RemoveRemoteOfferInput{
-			ModelName: modelName,
-			OfferURL:  *offer,
-		})
-		if len(errs) > 0 {
-			for _, v := range errs {
-				resp.Diagnostics.AddError("Client Error", v.Error())
-			}
-			return
-		}
-	} else {
-		err = r.client.Integrations.DestroyIntegration(&juju.IntegrationInput{
-			ModelName: modelName,
-			Endpoints: endpoints,
-		})
-		if err != nil {
-			resp.Diagnostics.AddError("Client Error", err.Error())
-			return
-		}
+	err = r.client.Integrations.DestroyIntegration(&juju.IntegrationInput{
```

Comment:
This change is moving from one extreme to another. Removing the offer without verifying if others are using it, to not removing at all. There should be a middle ground here. When destroying an integration using CMR, check to see if there are other users of it, if not, then remove it.

Reply:
Thanks for the suggestion. Implemented as per the suggesiton.